Lib/pathlib/types.py

```diff
@@ -240,45 +248,50 @@ class _ReadablePath(_JoinablePath):
 
     @property
     @abstractmethod
-    def info(self):
+    def info(self) -> PathInfo:
         """
         A PathInfo object that exposes the file type and other file attributes
         of this path.
         """
         raise NotImplementedError
 
     @abstractmethod
-    def __open_rb__(self, buffering=-1):
+    def __open_rb__(self, buffering: int = -1) -> BinaryIO:
         """
         Open the file pointed to by this path for reading in binary mode and
         return a file object, like open(mode='rb').
         """
         raise NotImplementedError
 
-    def read_bytes(self):
+    def read_bytes(self) -> bytes:
         """
         Open the file in bytes mode, read it, and close the file.
         """
         with magic_open(self, mode='rb', buffering=0) as f:
             return f.read()
 
-    def read_text(self, encoding=None, errors=None, newline=None):
+    def read_text(
+        self,
+        encoding: Optional[str] = None,
+        errors: Optional[str] = None,
+        newline: Optional[str] = None,
+    ) -> str:
         """
         Open the file in text mode, read it, and close the file.
         """
         with magic_open(self, mode='r', encoding=encoding, errors=errors, newline=newline) as f:
             return f.read()
 
     @abstractmethod
-    def iterdir(self):
+    def iterdir(self: _RP) -> Generator[_RP, None, None]:
```

Comment:
We can return an `Iterator` here. It doesn't matter that we return a generator I think.

Reply:
I'll make the change.